src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -608,6 +608,7 @@ def send_raw_request(cli_ctx, method, uri, headers=None, uri_parameters=None,  #
     try:
         r = requests.request(method, uri, params=uri_parameters, data=body, headers=headers,
                              verify=not should_disable_connection_verify())
+        logger.debug("Response Header : %s", r.headers if r else '')
```

Comment:
We can dump the headers regardless of `r.__bool__()` so that headers for non-successful response such as 404, 403 are also printed. I will create a new PR to address this.